File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.cpp
class Solution {
public:
    vector<int> deckRevealedIncreasing(vector<int>& deck) {
        sort(deck.begin(), deck.end(), greater<int>());
        vector<int> answer;

        for (auto num:deck) {
            if (answer.size() == 0) {
                answer.insert(answer.begin(), num);
            }
            else {
                rotate(answer.begin(), answer.end()-1, answer.end());
                answer.insert(answer.begin(), num);
            }
        }

        return answer;
    }
};

File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.py
class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        deck = sorted(deck)[::-1]
        answer = []

        for num in deck:
            if len(answer) == 0:
                answer.insert(0, num)
            else:
                answer.insert(0, answer[-1])
                answer = answer[:-1]
                answer.insert(0, num)
        
        return answer